### ISSUE TITLE
redundant null check doesn't recognize new pattern generated by javac in JDK 11

### ISSUE THREAD

**Issue by dougxc** [spotbugs/spotbugs#600]
When the code below is compiled by javac from a JDK 11 build that includes [JDK-8191708](https://bugs.openjdk.java.net/browse/JDK-8191708), SpotBugs generates a warning:
```
> java -jar ~/spotbugs-3.1.1/lib/spotbugs.jar -textui bin/Nullcheck.class
M C RCN: Nullcheck of os at line 9 of value previously dereferenced in Nullcheck.test(OutputStream)  At Nullcheck.java:[line 8]
Warnings generated: 1
```
This is not something that can be addressed changing the code so SpotBugs should detect this pattern and suppress warnings related to it.

```
import java.io.*;
import java.net.*;

public class Nullcheck {
    protected boolean test(OutputStream os) {
        try {
            try (OutputStream outputStream = os) {
                outputStream.write(0);
            }
            return true;
        } catch (IOException ex) {
        }
        return false;
    }
}
```

**Comment by dougxc** [comment]
Note that the `jdk-11-ea+5` build at http://jdk.java.net/11 does *not* include JDK-8191708 so to reproduce will require building the JDk from source or waiting for the next early access build to be made available.

**Comment by jlahoda** [comment]
Doug meant:
https://bugs.openjdk.java.net/browse/JDK-8194978

The patch changes how try-with-resources are desugared, the new output is much smaller than the JDK7/8 one. Please see the review thread for details on the desugaring (although it is not mentioning the exact bytecode, only desugared Java source code):
http://mail.openjdk.java.net/pipermail/compiler-dev/2018-March/011753.html

Should there be questions, I'll try to answer them as best as I can.

**Comment by dougxc** [comment]
Sorry for the wrong link - I fixed it in situ.

**Comment by JuditKnoll** [comment]
As far as I can see this was fixed in https://github.com/spotbugs/spotbugs/pull/1575.
